### DISCUSSION TITLE
Suggestion for a new package for asChild behavior

### DISCUSSION THREAD

**Discussion by alebelcor** [radix-ui/primitives#901] [Ideas]
Now that [Polymorphic](https://www.radix-ui.com/docs/primitives/utilities/polymorphic) and [Slot](https://www.radix-ui.com/docs/primitives/utilities/slot) are deprecated I was wondering if there might be any plans on publishing a new package that people could use to mimic the behavior of the `asChild` prop (along with prop & ref merging) in their components and/or design systems. If that makes sense.

I think that API turned out great and would love for more people to adopt it, even if they don't use Radix Primitives themselves (they should).

Thank you!

**Comment by jjenzz** [comment]
We have deprecated `Slot` mostly to prevent support requests but we use it ourselves still. So, if you wish to provide your own `asChild` implementation you can do the following:

```jsx
import { Slot } from '@radix-ui/react-slot';

type BoxProps = React.ComponentProps<'div'> & { 
	asChild?: boolean; 
};

const Box = (props: BoxProps) => {
	const { asChild, ...boxProps  } = props;
	const Comp = asChild ? Slot : 'div';
	return <Comp {...boxProps} />;
}
```

@peduarte perhaps we shouldn't deprecate it though if people will find it useful?
